drivers/adc/ad713x/ad713x.h

```diff
@@ -0,0 +1,878 @@
+#define AD713X_REG_TRANSFER_REGISTER		0x0F
+#define AD713X_REG_DEVICE_CONFIG1			0x10
+#define AD713X_REG_DATA_PACKET_CONFIG		0x11
+#define AD713X_REG_DIGITAL_INTERFACE_CONFIG	0x12
+#define AD713X_REG_POWER_DOWN_CONTROL		0x13
+#define AD713X_REG_AIN_RANGE_SELECT			0x14
+#define AD713X_REG_DEVICE_STATUS			0x15
+#define AD713X_REG_ODR_VAL_INT_LSB			0x16
+#define AD713X_REG_ODR_VAL_INT_MID			0x17
+#define AD713X_REG_ODR_VAL_INT_MSB			0x18
+#define AD713X_REG_ODR_VAL_FLT_LSB			0x19
+#define AD713X_REG_ODR_VAL_FLT_MID0			0x1A
+#define AD713X_REG_ODR_VAL_FLT_MID1			0x1B
+#define AD713X_REG_ODR_VAL_FLT_MSB			0x1C
+#define AD713X_REG_CHANNEL_ODR_SELECT		0x1D
+#define AD713X_REG_CHAN_DIG_FILTER_SEL		0x1E
+#define AD713X_REG_FIR_BW_SEL				0x1F
+#define AD713X_REG_GPIO_DIR_CTRL			0x20
+#define AD713X_REG_GPIO_DATA				0x21
+#define AD713X_REG_ERROR_PIN_SRC_CONTROL	0x22
+#define AD713X_REG_ERROR_PIN_CONTROL		0x23
+#define AD713X_REG_VCMBUF_CTRL				0x24
+#define AD713X_REG_DIAGNOSTIC_CONTROL		0x25
+#define AD713X_REG_MPC_CONFIG				0x26
+#define AD713X_REG_CH0_GAIN_LSB				0x27
+#define AD713X_REG_CH0_GAIN_MID				0x28
+#define AD713X_REG_CH0_GAIN_MSB				0x29
+#define AD713X_REG_CH0_OFFSET_LSB			0x2A
+#define AD713X_REG_CH0_OFFSET_MID			0x2B
+#define AD713X_REG_CH0_OFFSET_MSB			0x2C
+#define AD713X_REG_CH1_GAIN_LSB				0x2D
+#define AD713X_REG_CH1_GAIN_MID				0x2E
+#define AD713X_REG_CH1_GAIN_MSB				0x2F
+#define AD713X_REG_CH1_OFFSET_LSB			0x30
+#define AD713X_REG_CH1_OFFSET_MID			0x31
+#define AD713X_REG_CH1_OFFSET_MSB			0x32
+#define AD713X_REG_CH2_GAIN_LSB				0x33
+#define AD713X_REG_CH2_GAIN_MID				0x34
+#define AD713X_REG_CH2_GAIN_MSB				0x35
+#define AD713X_REG_CH2_OFFSET_LSB			0x36
+#define AD713X_REG_CH2_OFFSET_MID			0x37
+#define AD713X_REG_CH2_OFFSET_MSB			0x38
+#define AD713X_REG_CH3_GAIN_LSB				0x39
+#define AD713X_REG_CH3_GAIN_MID				0x3A
+#define AD713X_REG_CH3_GAIN_MSB				0x3B
+#define AD713X_REG_CH3_OFFSET_LSB			0x3C
+#define AD713X_REG_CH3_OFFSET_MID			0x3D
+#define AD713X_REG_CH3_OFFSET_MSB			0x3E
+#define AD713X_REG_MCLK_COUNTER				0x3F
+#define AD713X_REG_DIG_FILTER_OFUF			0x40
+#define AD713X_REG_DIG_FILTER_SETTLED		0x41
+#define AD713X_REG_INTERNAL_ERROR			0x42
+#define AD713X_REG_POWER_OV_ERROR_1			0x43
+#define AD713X_REG_POWER_UV_ERROR_1			0x44
+#define AD713X_REG_POWER_OV_ERROR_2			0x45
+#define AD713X_REG_POWER_UV_ERROR_2			0x46
+#define AD713X_REG_SPI_ERROR				0x47
+#define AD713X_REG_AIN_OR_ERROR				0x48
+#define AD713X_REG_AVDD5_VALUE				0x49
+#define AD713X_REG_DVDD5_VALUE				0x4A
+#define AD713X_REG_VREF_VALUE				0x4B
+#define AD713X_REG_LDOIN_VALUE				0x4C
+#define AD713X_REG_AVDD1V8_VALUE			0x4D
+#define AD713X_REG_DVDD1V8_VALUE			0x4E
+#define AD713X_REG_CLKVDD_VALUE				0x4F
+#define AD713X_REG_IOVDD_VALUE				0x50
+#define AD713X_REG_TEMPERATURE_DATA			0x51
+
+/*
+ * AD713X_REG_INTERFACE_CONFIG_A
+ */
+#define AD713X_INT_CONFIG_A_SOFT_RESET_MSK			BIT(7)
+#define AD713X_INT_CONFIG_A_SOFT_RESET_MODE(x)		(((x) & 0x1) << 7)
```

Comment:
if a define is limited to a bit i would no do a MSK and MODE just 
<code>
#define AD713X_INT_CONFIG_A_SOFT_RESET			BIT(7)

val |= AD713X_INT_CONFIG_A_SOFT_RESET;
val &= ~AD713X_INT_CONFIG_A_SOFT_RESET;
</code>